src/main/java/Quill.java

```diff
@@ -0,0 +1,127 @@
+import quill.task.Deadline;
+import quill.task.Event;
+import quill.task.Task;
+import quill.task.Todo;
+
+import java.util.Scanner;
+public class Quill {
+    public static final String horizontalLine = "____________________________________________________________";
+    public static void printAddTask(Task tasks) {
+        System.out.println(horizontalLine + "\nGot it. I've added this task:");
+        System.out.println(tasks.toString());
+        System.out.println("Now you have " + Task.getTotalTasks() + " tasks in the list.\n" + horizontalLine);
+    }
+
+    public static void performMarkOrUnmark(String line, Task[] tasks, boolean isDone) {
+        try {
+            int taskNumber = Integer.parseInt(line) - 1;
+            if (isDone) {
+                tasks[taskNumber].markAsDone();
+                System.out.println(horizontalLine + "\nNice! I've marked this task as done:");
+            } else {
+                tasks[taskNumber].markAsNotDone();
+                System.out.println(horizontalLine + "\nOK, I've marked this task as not done yet:");
+            }
+            System.out.println(tasks[taskNumber].toString() + "\n" + horizontalLine);
+        } catch (NullPointerException e) {
+            System.out.println("Hey, wake up! That task? Non-existent. Try something real.");
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("1 to 100 only. Seriously, no zeros, no hundreds. Got it?");
+        } catch (NumberFormatException e) {
+            System.out.println("Listen up! Numbers only, got it? Don't bother with anything else");
+        }
+    }
+
+    public static void main(String[] args) throws QuillException {
+        String name = "Quill";
+        int MAX_TASKS = 100;
+        String line;
+        Task[] tasks = new Task[MAX_TASKS];
+        Scanner in = new Scanner(System.in);
+
+        System.out.println(horizontalLine + "\nHello! I'm " + name + ".");
+        System.out.println("What can i do for you?\n" + horizontalLine);
+
+        line = in.nextLine();
+
+        while(true) {
+            String command;
+            int taskNumber = Task.getTotalTasks();
+            int index = line.indexOf(" ");
+            if (index != -1) {
+                command = line.substring(0, index);
+                line = line.substring(index + 1);
+            } else {
+                command = line;
+                line = "";
+            }
+            switch(command) {
+            case "bye":
+                System.out.println(horizontalLine + "\nBye! Hope to see you again soon!\n" + horizontalLine);
+                return;
+            case "list":
+                if (Task.isEmpty()) {
+                    System.out.println(horizontalLine + "\nZero tasks. Add something already.");
+                } else {
+                    System.out.println(horizontalLine + "\nHere are the tasks in your list:");
+                    for (int i = 0; i < Task.getTotalTasks(); i++) {
```

Comment:
You could consider refactoring this code to avoid deep nesting and adhere to this code quality [guideline](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting)